runtime/src/identity.rs

```diff
@@ -5,42 +5,35 @@ use std::{
 };
 
 use anyhow::Result;
-#[cfg(target_env = "sgx")]
 use rand::{rngs::OsRng, Rng};
-#[cfg(target_env = "sgx")]
-use sgx_isa::Report;
-use sgx_isa::Targetinfo;
+use sgx_isa::{Report, Targetinfo};
 use thiserror::Error;
-#[cfg(target_env = "sgx")]
 use tiny_keccak::{Hasher, TupleHash};
 
-#[cfg_attr(not(target_env = "sgx"), allow(unused))]
-use crate::common::crypto::hash::Hash;
 use crate::{
     common::{
         crypto::{
+            hash::Hash,
             mrae::deoxysii::{self, Opener},
             signature::{self, Signature, Signer},
             x25519,
         },
-        sgx::{EnclaveIdentity, Quote, QuotePolicy, VerifiedQuote},
+        sgx::{self, EnclaveIdentity, Quote, QuotePolicy, VerifiedQuote},
         time::insecure_posix_time,
     },
     consensus::registry::EndorsedCapabilityTEE,
 };
 
 /// Context used for computing the RAK digest.
-#[cfg_attr(not(target_env = "sgx"), allow(unused))]
 const RAK_HASH_CONTEXT: &[u8] = b"oasis-core/node: TEE RAK binding";
 /// Context used for deriving the nonce used in quotes.
-#[cfg_attr(not(target_env = "sgx"), allow(unused))]
 const QUOTE_NONCE_CONTEXT: &[u8] = b"oasis-core/node: TEE quote nonce";
 
 /// A dummy RAK seed for use in non-SGX tests where integrity is not needed.
-#[cfg(not(target_env = "sgx"))]
+#[cfg(not(any(target_env = "sgx", feature = "debug-mock-sgx")))]
```

Comment:
Do we panic (or prevent) anywhere if `target_env = "sgx"` and `feature = "debug-mock-sgx"`?

Reply:
Good point, will add a check in `lib.rs`.